docs/data-sources/image.md

```diff
@@ -0,0 +1,53 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "docker_image Data Source - terraform-provider-docker"
+subcategory: ""
+description: |-
+  docker_image provides details about a specific Docker Image which need to be presend on the Docker Host
+---
+
+# docker_image (Data Source)
+
+`docker_image` provides details about a specific Docker Image which need to be presend on the Docker Host
+
+## Example Usage
+
+```terraform
+# uses the 'latest' tag
+data "docker_image" "latest" {
+  name = "nginx"
+}
+
+# uses a specific tag
+data "docker_image" "specific" {
+  name = "nginx:1.17.6"
+}
+
+# use the image digest
+data "docker_image" "digest" {
+  name = "nginx@sha256:36b74457bccb56fbf8b05f79c85569501b721d4db813b684391d63e02287c0b2"
+}
+
+# uses the tag and the image digest
+data "docker_image" "tag_and_digest" {
+  name = "nginx:1.19.1@sha256:36b74457bccb56fbf8b05f79c85569501b721d4db813b684391d63e02287c0b2"
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- **name** (String) The name of the Docker image, including any tags or SHA256 repo digests.
+
+### Optional
+
+- **id** (String) The ID of this resource.
+
+### Read-Only
+
+- **latest** (String, Deprecated) The ID of the image in the form of `sha256:<hash>` image digest. Do not confuse it with the default `latest` tag.
+- **sha256_digest** (String) The image sha256 digest in the form of `sha256:<hash>`.
```

Comment:
Having a `repo_digest` string would probably be the most useful, rather than repeating the `id` value again, which as @suzuki-shunsuke points out is not an actual image digest (but the value does appear to be stable between two different machines).